src/txmempool.cpp

```diff
@@ -904,3 +904,25 @@ void CTxMemPool::TrimToSize(size_t sizelimit) {
         RemoveStaged(stage, false);
     }
 }
+
+// BU: begin
+void CTxMemPool::UpdateTransactionsPerSecond()
+{
+    std::lock_guard<std::mutex> lock(cs_txPerSec);
+
+    static int64_t nLastTime = GetTime();
```

Comment:
For consistency, I suggest either make this a member variable, or make all the variables global statics.

For reusability, consider that doesn't really need anything from the mempool. This could be a generic class `RateCounter` and mempool could hold an instance of it (`RateCounter txsPerSecond`)

ATMP can call it via `mempool.GetTxsPerSecond().Tick()` and retrieve current value as `mempool.GetTxsPerSecond().Value()`

Reply:
https://github.com/BitcoinUnlimited/BitcoinUnlimited/pull/1514 fixes this nicely. I will do it in another PR that includes the base stats tracking.